### PR TITLE
Home screen fix

### PR DESCRIPTION
**Pull request by HauntedMilkshake**
- Filtration to home screen workouts from last month

Previously we didn't filter workouts whenever calculating where to place them on chart and workouts from previous months 
would be counted which led to a crash between months with different day numbers

We also would trim the last week of the month if that month had more than 4 weeks for some reason.

### CODE REVIEW

**Review by trifonov-i — APPROVED** [review]

File: app/src/main/java/bg/zahov/app/data/provider/WorkoutProviderImpl.kt [hunk 1]
```diff
@@ -111,6 +112,14 @@ class WorkoutProviderImpl : WorkoutProvider {
         workoutRepo.updateTemplateWorkout(workoutId, date, newExercises)
     }
 
+    override suspend fun getPastWorkoutsForCurrentMonthWithWorkoutCount(): Flow<Pair<List<Workout>, Int>> {
```
Comment (dimitarstoyanovtumba) [inline app/src/main/java/bg/zahov/app/data/provider/WorkoutProviderImpl.kt:115]: No need for such a long name. Most of this information and more if needed can be added to the kdoc.

File: app/src/main/java/bg/zahov/app/data/interfaces/WorkoutProvider.kt [hunk 2]
```diff
@@ -30,6 +30,7 @@ interface WorkoutProvider {
         date: LocalDateTime,
         newExercises: List<Exercise>,
     )
+    suspend fun getPastWorkoutsForCurrentMonthWithWorkoutCount(): Flow<Pair<List<Workout>, Int>>
```
Comment (dimitarstoyanovtumba) [inline app/src/main/java/bg/zahov/app/data/interfaces/WorkoutProvider.kt:33]: Needs kdoc.

File: app/src/main/java/bg/zahov/app/data/provider/WorkoutProviderImpl.kt [hunk 3]
```diff
@@ -111,6 +112,14 @@ class WorkoutProviderImpl : WorkoutProvider {
...
+        return workoutRepo.getPastWorkouts()
+            .mapNotNull {
+                Pair(it.filter { workout -> workout.date.month == LocalDate.now().month },
+                    it.size)
```
[the remark this answers is not in the document]
Comment (HauntedMilkshake) [reply app/src/main/java/bg/zahov/app/data/provider/WorkoutProviderImpl.kt:119]: The screen shows the number of all workouts the user has performed, while the function filters out only the workouts from the last month.

File: app/src/main/java/bg/zahov/app/data/interfaces/WorkoutProvider.kt (hunk 2 above)
Comment (HauntedMilkshake) [reply app/src/main/java/bg/zahov/app/data/interfaces/WorkoutProvider.kt]: 👍

File: app/src/main/java/bg/zahov/app/data/provider/WorkoutProviderImpl.kt (hunk 1 above)
Comment (HauntedMilkshake) [reply app/src/main/java/bg/zahov/app/data/provider/WorkoutProviderImpl.kt]: 👍

File: app/src/main/java/bg/zahov/app/data/provider/WorkoutProviderImpl.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (dimitarstoyanovtumba) [reply app/src/main/java/bg/zahov/app/data/provider/WorkoutProviderImpl.kt:128]: The map and filter functions are okay. I think you don't need to map to a Pair. You can simplify the code by mapping to `List<Workout>`.

[the remark this answers is not in the document]
Comment (HauntedMilkshake) [reply app/src/main/java/bg/zahov/app/data/provider/WorkoutProviderImpl.kt:128]: 👍

File: app/src/main/java/bg/zahov/app/data/provider/WorkoutProviderImpl.kt [hunk 4]
```diff
@@ -112,6 +113,21 @@ class WorkoutProviderImpl : WorkoutProvider {
         workoutRepo.updateTemplateWorkout(workoutId, date, newExercises)
     }
 
+    /**
+     * Retrieves a flow of workouts from the current month along with their count.
+     *
+     * This function filters the workouts to include only those that occurred in the
+     * current month.
+     *
+     * @return A [Flow] emitting a [Pair] of a list of [Workout] objects and the
```
Comment (dimitarstoyanovtumba) [inline app/src/main/java/bg/zahov/app/data/provider/WorkoutProviderImpl.kt:122]: nit: documentation can be updated

**Pull request merged**